src/ILLink.RoslynAnalyzer/RequiresISymbolExtensions.cs

```diff
@@ -51,6 +51,11 @@ public static bool IsOverrideInRequiresScope (this ISymbol member, string requir
 
 		private static bool IsInRequiresScope (this ISymbol member, string requiresAttribute, bool checkAssociatedSymbol)
 		{
+			// Requires attribute on a type does not silence warnings that originate
+			// from the type directly.
+			if (member is ITypeSymbol)
+				return false;
+
 			if (member is ISymbol containingSymbol) {
```

Comment:
I think this line makes no sense, we know that the member is ISymbol, because that's what we get as input so we are effectively only changing the name to containingSymbol, but is not the containingSymbol, so I guess we went off track at some point.

Reply:
I think it was there as a null check. I am not sure if it's necessary but I will keep it to be safe. I made the null check more explicit to avoid just renaming the variable.

Reply:
I think it was there as a null check. Not sure how necessary it is, but I'd rather keep it. I changed the code to make it more clear and to avoid pointlessly changing the variable name. Thanks!

Reply:
But we have nullable enabled, the ISymbol cannot be null. Also Im not sure if calling it containingSymbol, the fact that this is a containingSymbol or not is completely on the caller.

Reply:
I was a little worried that we could be getting a null ISymbol from a null-oblivious API somewhere, especially since this logic has caused nullref exceptions in the past. But I looked at all the callers and I think it's ok, so I removed the null check.

For the name, I changed it to just `symbol`. `member` doesn't seem right since it can be called on types. Then again, the method doesn't make sense for all kinds of `ISymbol`, so I don't really have a strong opinion - let me know if you prefer.

Reply:
When I was doing the reflection call GetMembers() nested types is one of the things that returns, so I consider a type as a member (see https://docs.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo?view=net-6.0)

Reply:
Changed to `member`